### PR TITLE
TYP: handle mypy ignore in pandas/_testing

### PR DESCRIPTION
**Pull request by ivanovmg**
- [ ] xref #37715
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Handle mypy ignore comments in ``pandas/_testing.py``, function ``write_to_compressed``.

### CODE REVIEW

**Comment by ivanovmg** [comment]
It's green, except failing ``TestHDFStore.test_append_with_data_columns`` on Linux py39.

File: pandas/_testing.py [hunk 1]
```diff
@@ -299,39 +309,35 @@ def write_to_compressed(compression, path, data, dest="test"):
...
-    elif compression == "bz2":
-        # pandas\_testing.py:290: error: Incompatible types in assignment
-        # (expression has type "Type[BZ2File]", variable has type
-        # "Type[ZipFile]")
-        compress_method = bz2.BZ2File  # type: ignore[assignment]
-    elif compression == "xz":
-        compress_method = get_lzma_file(lzma)
-    else:
-        raise ValueError(f"Unrecognized compression type: {compression}")
+    compress_method = _select_compress_method(compression)
```
Comment (jreback) [inline pandas/_testing.py:312]: i think you could leave this inline and just type compress_method no?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_testing.py (hunk 1 above)
Comment (ivanovmg) [reply pandas/_testing.py:312]: I guess so.
I just thought that some small function extraction will be helpful here.
But if you suggest against the extraction, I can definitely do that way as well.

Comment (WillAyd) [reply pandas/_testing.py:312]: I would prefer for this to stay inline as well

Comment (ivanovmg) [reply pandas/_testing.py]: I inlined the function and also slid the conditionals. Can revert if this is not what you expect, but in my opinion, it simplifies the understanding that ``zip`` has a different signature as compared to other compression methods.

**Review by jreback — APPROVED** [review]

**Pull request merged**